server/src/Irony.cpp

```diff
@@ -428,12 +471,28 @@ void Irony::candidates() const {
       // https://github.com/Sarcasm/irony-mode/pull/78#issuecomment-37115538
       if (chunkKind == CXCompletionChunk_TypedText && !typedTextSet) {
         typedtext = chunk.text();
+        if (!startsWith(typedtext, prefix, caseInsensitive)) {
+          hasPrefix = false;
+          break;
+        }
         // annotation is what comes after the typedtext
         annotationStart = prototype.size();
         typedTextSet = true;
       }
     }
 
+    if (!hasPrefix) {
+      continue;
+    }
+    if (!typedTextSet) {
```

Comment:
Oh I see, this information could actually be useful to provide user hints, e.g. with eldoc.